File: sentimental-credit/credit.py
# TODO
from cs50 import get_string


def main():
    # get the card number from the user
    cardNum = get_string("Number: ")

    # check the length of the card
    cardLen = len(str(cardNum))

    evenSum = 0
    oddSum = 0
    checkSum = 0

    # if the card number has the correct amount of numbers
    if cardLen != 13 and cardLen != 15 and cardLen != 16:
        print("INVALID")
        exit()

    for i in range(-2, (-cardLen-1), -2):

        tempMul = int(cardNum[i]) * 2

        if tempMul >= 10:
            evenSum = evenSum + tempMul - 9
        else:
            evenSum = evenSum + tempMul

    for i in range(-1, (-cardLen-1), -2):
        tempOdd = int(cardNum[i])
        oddSum = oddSum + tempOdd

    # total up so this can be checked for which card the number is
    checkSum = evenSum + oddSum

    if checkSum % 2 == 0:
        # Check for american express
        if int(cardNum[0]) == 3 and int(cardNum[1]) == 4 or int(cardNum[1]) == 7:
            print("AMEX")

        # check for mastercard
        elif int(cardNum[0]) == 5 and int(cardNum[1]) == 1 or int(cardNum[1]) == 2 or int(cardNum[1]) == 3 or int(cardNum[1]) == 4 or int(cardNum[1]) == 5:
            print("MASTERCARD")

        # check for visa
        elif int(cardNum[0]) == 4:
            print("VISA")

        # if the number matches none of the above then it is invalid
        else:
            print("INVALID")


# Call the main function
main()

File: credit/credit.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    long long n, odd, even;

    do
    {
        //get the users input
        n = get_long("please enter your card number ");
    }
    while (n < 0);
    {
        //set the variables to be used
        int checksum, evennumber, eventotal, oddtotal;

        //loop around to get the total for the even numbers
        for (even = n / 10, eventotal = 0; even > 0; even /= 100)
        {
            evennumber = even % 10;

            if ((2 * evennumber) > 9)
            {
                eventotal += (2 * evennumber) / 10;
                eventotal += ((2 * evennumber) % 10);
            }
            else
            {
                eventotal += ((2 * evennumber) % 10);
            }
        }

        //loop around to get the total for the odd numbers
        for (odd = n, oddtotal = 0; odd > 0; odd /= 100)
        {
            oddtotal += odd % 10;
        }

        checksum = eventotal + oddtotal;

        if (checksum % 10 == 0)
        {
            //Check for American Express
            if ((n >= 340000000000000 && n <= 350000000000000) || (n >= 370000000000000 && n <= 380000000000000))
            {
                printf("AMEX\n");
            }
            //Check for Mastercard
            else if (n >= 5100000000000000 && n <= 5600000000000000)
            {
                printf("MASTERCARD\n");
            }
            //check for visa
            else if ((n >= 4000000000000 && n <= 4200000000000) || (n >= 4000000000000000 && n <= 4200000000000000) || (n == 4222222222222))
            {
                printf("VISA\n");
            }
            //if none of the above card is invalid
            else
            {
                printf("INVALID\n");
            }
        }
        else
        {
            printf("INVALID\n");
        }
    }
}